akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -257,7 +257,12 @@ private[io] class SelectionHandler(settings: SelectionHandlerSettings) extends A
                               decision: SupervisorStrategy.Directive): Unit =
         try {
           val logMessage = cause match {
-            case e: ActorInitializationException if e.getCause ne null ⇒ e.getCause.getMessage
+            case e: ActorInitializationException if (e.getCause ne null) && (e.getCause.getMessage ne null) ⇒ e.getCause.getMessage
+            case e: ActorInitializationException if (e.getCause ne null) ⇒
+              e.getCause match {
```

Comment:
Perhaps.  In my projects I tend to have a generic facility for turning exceptions into meaningful messages because, unfortunately, the "standard" for this is shoddy.  It would be awesome if Akka had such a facility that covered the JVM and Akka exceptions because useful log messages are one of the strongest aids for debugging complex asynch programs that do not lend themselves well to using debuggers. I'd be willing to help make this improvement to Akka but it is a bit of a wide-sweeping change. This patch just aimed at taking care of one specific instance where it is important but if the Akka developers want this generalized, I'd be willing to help.